src/generators.rs

```diff
@@ -181,6 +181,32 @@ impl BulletproofGens {
         }
     }
 
+    /// Precomputes additional generators and increases the capacity.
+    pub fn add_gens(&mut self, add_gens: usize) {
+        use byteorder::{ByteOrder, LittleEndian};
+
+        for i in 0..self.party_capacity {
+            let party_index = i as u32;
+            let mut label = [b'G', 0, 0, 0, 0];
+            LittleEndian::write_u32(&mut label[1..5], party_index);
+            self.G_vec[i].append(
+                &mut GeneratorsChain::new(&label)
+                    .skip(self.gens_capacity)
```

Comment:
In the current form this `.skip` is expensive - it generates all the intermediate points. To avoid this cost we need either of two things:

1. Store `GeneratorsChain` inside each item of `G_vec/H_vec` and continue iterating from where it stopped.
2. Modify `GeneratorsChain` to return some lazily-computed points: e.g. yield `GeneratedPoint([u8;64])` that implements `Into<RistrettoPoint>`. This seems to be cleaner. Then, `skip` becomes cheap.